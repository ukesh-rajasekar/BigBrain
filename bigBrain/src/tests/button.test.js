import React from 'react'
import { render } from '@testing-library/react';
import { shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../components/button'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });



describe('Render Button', () => {
  it('render with minimal props', () => {  
    const ButtonComponent = renderer.create(<Button />).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });

  test('uses custom title', () => {
    const buttonText = 'Start'
    const ButtonComponent = render(<Button buttonText = {buttonText}/>)
    const text = ButtonComponent.getByText(buttonText)
    expect(text).toBeInTheDocument;
});

test('uses custom name', () => {
  const buttonText = 'Stop'
  const ButtonComponent = render(<Button buttonText = {buttonText}/>)
  const text = ButtonComponent.getByText(buttonText)
  expect(text).toBeInTheDocument;
});

  it('triggers onClick event handler when clicked', () => {
    const buttonAction = jest.fn();
    shallow(<Button buttonAction={buttonAction}/>).simulate('click');
    expect(buttonAction).toHaveBeenCalledTimes(1);
  })
 
  
  });
