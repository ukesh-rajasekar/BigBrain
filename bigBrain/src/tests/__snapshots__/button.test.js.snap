// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button render with minimal props 1`] = `
<button
  className="btn btn-primary btn-sm"
  disabled={false}
  type="button"
>
    
</button>
`;
